### DISCUSSION TITLE
魏老师您好，我在MAC系统下安装fealpy时出现了numpy版本过低的问题

### DISCUSSION THREAD

**Discussion by wangjk97** [weihuayi/fealpy#45] [Installation]
当我安装fealpy时
kail@wangjunkaideMacBook-Pro ~ % cd fealpy                       
kail@wangjunkaideMacBook-Pro fealpy % sudo python setup_mac.py develop
Password:
running develop
running egg_info
writing requirements to fealpy.egg-info/requires.txt
writing fealpy.egg-info/PKG-INFO
writing top-level names to fealpy.egg-info/top_level.txt
writing dependency_links to fealpy.egg-info/dependency_links.txt
reading manifest file 'fealpy.egg-info/SOURCES.txt'
writing manifest file 'fealpy.egg-info/SOURCES.txt'
running build_ext
Creating /Library/Python/2.7/site-packages/fealpy.egg-link (link to .)
fealpy 1.0 is already the active version in easy-install.pth

Installed /Users/kail/fealpy
Processing dependencies for fealpy==1.0
error: numpy 1.8.0rc1 is installed but numpy<2.0,>=1.10 is required by set(['pyfftw'])

我尝试更新了numpy到最新版本

kail@wangjunkaideMacBook-Pro fealpy % sudo pip install -U numpy
WARNING: The directory '/Users/kail/Library/Caches/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
Collecting numpy
  Downloading numpy-1.19.1-cp38-cp38-macosx_10_9_x86_64.whl (15.3 MB)
     |████████████████████████████████| 15.3 MB 15 kB/s 
Installing collected packages: numpy
  Attempting uninstall: numpy
    Found existing installation: numpy 1.19.0
    Uninstalling numpy-1.19.0:
      Successfully uninstalled numpy-1.19.0
Successfully installed numpy-1.19.1

然后将默认python改为3.8

kail@wangjunkaideMacBook-Pro ~ % python                
Python 3.8.2 (v3.8.2:7b3ab5921f, Feb 24 2020, 17:52:18) 

但是还是出现相同报错，没办法安装，好像是我的安装的时候还是在mac自带的python2.7下安装的(个人感觉下面是我的报错原因)
Creating /Library/Python/2.7/site-packages/fealpy.egg-link (link to .)
fealpy 1.0 is already the active version in easy-install.pth
请问一下魏老师这个该如何解决呢？

**Comment by weihuayi** [comment]
你用的 Python 版本不对，是 2.7。 你如果装了 python3.8， 有没有装 pip3？ 这个两个调用命令可能是 python3 和 pip3。
&nbsp;
&nbsp;
------------------&nbsp;Original&nbsp;------------------

**Comment by wangjk97** [comment]
老师，我的python3和pip3都是安装好了的

kail@wangjunkaideMacBook-Pro ~ % pip3 -V
pip 20.2.1 from /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pip (python 3.8)
kail@wangjunkaideMacBook-Pro ~ % python3 -V
Python 3.8.2

但是安装fealpy时还是会出现之前相同的报错

**Comment by yoczhang** [comment]
从你提供的信息看
> error: numpy 1.8.0rc1 is installed but numpy<2.0,>=1.10 is required by set(['pyfftw'])

这里要求 `pyfftw` 要求的是 `numpy<2.0,>`.
而你通过 `sudo pip install -U numpy` 升级后
> Uninstalling numpy-1.19.0:
 Successfully uninstalled numpy-1.19.0
 Successfully installed numpy-1.19.1

你只是将 `numpy` 升级到了 ` numpy-1.19.1`, 你可以再上网查一下安装最新版本 `numpy` 的方法, 安装最新的 `numpy`.

**Comment by wangjk97** [comment]
> 从你提供的信息看
> 
> > error: numpy 1.8.0rc1 is installed but numpy<2.0,>=1.10 is required by set(['pyfftw'])
> 
> 这里要求 `pyfftw` 要求的是 `numpy<2.0,>`.
> 而你通过 `sudo pip install -U numpy` 升级后
> 
> > Uninstalling numpy-1.19.0:
> > Successfully uninstalled numpy-1.19.0
> > Successfully installed numpy-1.19.1
> 
> 你只是将 `numpy` 升级到了 ` numpy-1.19.1`, 你可以再上网查一下安装最新版本 `numpy` 的方法, 安装最新的 `numpy`.

您好，我之前也发现了这个问题，可是当我卸载numpy之后使用pip3重新安装numpy时，始终默认安装的numpy 1.19.1
而我在numpy官网也只看到了1.19.1的版本发布

kail@wangjunkaideMacBook-Pro ~ % pip3 install numpy                            
Collecting numpy
  Downloading numpy-1.19.1-cp38-cp38-macosx_10_9_x86_64.whl (15.3 MB)
     |████████████████████████████████| 15.3 MB 13 kB/s 
Installing collected packages: numpy
Successfully installed numpy-1.19.1

**Comment by yoczhang** [comment]
好吧, 我可能搞错了😂

**Comment by weihuayi** [comment]
你先把 python2.7 下的 fealpy 清理掉。再按照 Ubuntu 的流程装一遍，有问题，把相应的信息贴到这里来
